test/__mocks__/TemplateExtension.ts

```diff
@@ -0,0 +1,11 @@
+export const mockExecute = jest.fn();
```

Comment:
That's interesting. I actually never used Jest mocks before but I wonder, isn't it an over complex solution? Why not just do this instead:

```javascript
export class DummyExtension {

    public executeCalled = 0;

    public execute() {
        this.executeCalled++;
    }
}
```

Then use it:

```javascript
const firstTemplateExtension: TemplateExtension = (new DummyExtension() as TemplateExtension);
const secondTemplateExtension: TemplateExtension = (new DummyExtension() as TemplateExtension);
```

It's less "magical"...

P.S. I do use fakes and mocks in strong typed languages but don't see the need since JavaScript is dynamic.

Reply:
I tend to use jest mocks all the time as they have good functionality so I don't have to code it as per the example above, especially when testing execution it gives access to things like toHaveBeenCalled().

Not averse to changing it.

Reply:
However I have a compromise solution which uses a class as above but rather than using an explicit execute I use:

```js
const beforeCompilationExecute = jest.fn();
const afterCompilationExecute = jest.fn();
const setUtilities = jest.fn();

class BeforeCompilationExtension extends TemplateExtension {
    public execute = beforeCompilationExecute;
    public setUtilities = setUtilities;
}

class AfterCompilationExtension extends TemplateExtension {
    public execute = afterCompilationExecute;
    public setUtilities = setUtilities;
}
```

Thus allowing me to use tests like:

```js
expect(setUtilities).toHaveBeenCalledTimes(2);
expect(beforeCompilationExecute).toHaveBeenCalledTimes(1);
expect(afterCompilationExecute).toHaveBeenCalledTimes(1);
```